aloha-core/src/test/scala/com/eharmony/aloha/models/multilabel/MultilabelModelTest.scala

```diff
@@ -47,124 +63,336 @@ class MultilabelModelTest extends ModelSerializationTestHelper {
+
+    val report = reportNoPredictionPartial(labelsAndInfoMissingLabels)
+    assertEquals(Vector(NoLabelsError) ++ errorMessages, report.audited.errorMsgs)
+    assertEquals(None, reportNoPredictionEmpty.audited.value)
+    assertEquals(Set(), report.audited.missingVarNames)
   }
 
   @Test def testReportPredictorError(): Unit = {
-    // Make sure Subvalue.natural == None
-    // Check the values of Subvalue.audited and make sure they are as expected.
-    // Subvalue.audited.value should be None.  Check the errors and missing values.
-
-    fail()
+    val (throwable, stackTrace) = getThrowable("error")
+    val missingVariables = Seq("a", "b")
+    val report = reportPredictorError(
+      ModelId(),
+      labelsAndInfoEmpty.copy(labelsNotInTrainingSet = labelsNotInTrainingSet),
+      scm.Map("" -> missingVariables),
+      throwable,
+      Auditor
+    )
+    assertEquals(Vector(stackTrace) ++ errorMessages, report.audited.errorMsgs)
+    assertEquals(missingVariables.toSet, report.audited.missingVarNames)
+    assertEquals(None, report.natural)
+    assertEquals(None, report.audited.value)
   }
 
   @Test def testReportSuccess(): Unit = {
-    // Make sure Subvalue.natural == Some(value)
-    // Check the values of Subvalue.audited and make sure they are as expected.
-    // Subvalue.audited.value should be Some(value2).
-    // 'value' should equal 'value2'.
-    // Check the errors and missing values.
+    val predictions = Map("label" -> 1.0)
+    val report   = reportSuccess(
+      modelId    = ModelId(),
+      labelInfo  = labelsAndInfoEmpty.copy(labelsNotInTrainingSet = labelsNotInTrainingSet),
+      missing    = scm.Map("" -> labelsNotInTrainingSet),
+      prediction = predictions,
+      auditor    = Auditor
+    )
+    assertEquals(Some(predictions), report.natural)
+    assertEquals(Some(predictions), report.audited.value)
+    assertEquals(report.natural, report.audited.value)
+  }
+
+  @Test def testLabelsForPredictionContainsProblemsWhenNoLabelProvided(): Unit = {
+    val labelsAndInfoNoLabels = labelsForPrediction(
+      example          = Map[Label, Label](),  // no label provided
+      labelsOfInterest = labelsOfInterestExtractor,
+      labelToInd       = labelsInTrainingSetToIndex)
 
-    fail()
+    // In this scenario no problems are found but an
+    // Option(GenAggFuncAccessorProblems) is returned instead of None
+    val problemsNoLabelsExpected = Option(GenAggFuncAccessorProblems(Seq(), Seq()))
+    assertEquals(problemsNoLabelsExpected, labelsAndInfoNoLabels.problems)
   }
 
-  @Test def testLabelsForPredictionContainsProblemsWhenLabelsIsEmpty(): Unit = {
-    // Test this:
-    //    val problems =
-    //      if (labelsShouldPredict.nonEmpty) None
-    //      else Option(labelsOfInterest.accessorOutputProblems(a))
+  @Test def testLabelsForPredictionContainsProblemsWhenLabelsIsNotPresent(): Unit = {
+    val labelsExtractor =
+      (label: Map[Label, Label]) => label.get("labels") match {
```

Comment:
This can't be right.  Look at the types.  Looks like you want `labelsExtractor` to have the following type:

```scala
val labelsExtractor: Map[String, String] => Option[sci.IndexedSeq[String]] = ???
```

But then, think about what you're trying to do. `label.get("labels")` must return a `Option[String]`, based on the API.  So, you are checking if an `Option[String]` is a `sci.IndexedSeq[_]`.  Then if it is, you're wrapping it in another Option.  I don't understand.  First of all:

```scala
Option.empty[String].isInstanceOf[sci.IndexedSeq[_]]   // false
Option("").isInstanceOf[sci.IndexedSeq[_]]             // false
```

It looks like you are trying to subvert the type system but I don't see the benefit.  It's in instances like this that the compiler warns about erasure when you do `case ls: sci.IndexedSeq[_] =>` because this is probably a bad check.  This should probably be something like the following instead: 

```scala
(label: Map[Label, Label]) => label.get("labels") match {
  // strVal is a string.  Do something with it to create a Option[sci.IndexedSeq[String]].
  case Some(strVal) => ??? 
  case None => None
}
```

Reply:
Which means that this could probably just be done with a `flatMap` (or `map`) after calling `label.get`.